scripts/core/memory.yml

```diff
@@ -540,3 +540,87 @@ type: class
 desc: "C++ wrapper for memory allocation"
 name: $xMem
 owner: $xContext
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "atomic access attribute flags"
+class: $xMem
+name: $x_memory_atomic_attr_exp_flags_t
+version: "1.7"
+etors:
+    - name: NONE
```

Comment:
this is too coarse and prevents atomics both on host & device.
We need separate controls for no atomics on host & device

Reply:
@MichalMrozek : could you elaborate? didnt understand the part of needing to separate. 
These are flags, not enums, so they could be passed as combination. So for instance:

No atomics anywhere: NONE

No atomics on host, but atomics on device: DEVICE_ATOMIC

Atomics on host, but no atomics on device: DEVICE_ATOMIC

Atomics on host and device: HOST_ATOMIC | DEVICE_ATOMIC

because of the last case, dont know if we actually need SYSTEM_ATOMIC.

Reply:
This doesn't sound from the current description.
By default all shared allocations support atomics / concurrent atomics on both host and device.

Now when you use this API with DEVICE_ATOMIC on such shared allocation you tell the driver that atomics may be performed on the device, but it doesn't tell anything that they are prohibited on the host.

I don't see mapping that you explained in comment above, to what this extensions documents in description, I don't see that when you specify DEVICE_ATOMIC, then it means no atomics on the host.

Only NONE flag disallows atomics, but it does it globally.

Reply:
no_atomics - no atomics performed from device & system
no_host_atomics - no atomics performed from host
no_device_atomics - no atomics performed from device
no_system_atomics - no atomics performed concurrently from host & device
host_atomics - host may do the atomics
device_atomics - device may do the atomics
system_atomics - host & device may do concurrent system atomics, requires support from shared USM cap

Reply:
@MichalMrozek thanks. please take a look.